pkg/reconciler/pingsource/controller/pingsource.go

```diff
@@ -192,6 +211,31 @@ func (r *Reconciler) createReceiveAdapter(ctx context.Context, src *v1alpha1.Pin
 	return ra, nil
 }
 
+func (r *Reconciler) reconcileJobRunner(ctx context.Context, source *v1alpha1.PingSource) (*appsv1.Deployment, error) {
+	d, err := r.deploymentLister.Deployments(system.Namespace()).Get(jobRunnerName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			args := resources.JobRunnerArgs{
+				ServiceAccountName: jobRunnerName,
+				JobRunnerName:      jobRunnerName,
+				JobRunnerNamespace: system.Namespace(),
+				Image:              r.jobRunnerImage,
+			}
+			expected := resources.MakeJobRunner(args)
+			d, err := r.KubeClientSet.AppsV1().Deployments(system.Namespace()).Create(expected)
+
+			if err != nil {
+				r.Recorder.Eventf(source, corev1.EventTypeWarning, pingSourceDeploymentCreated, "Cluster-scoped deployment not created (%v)", err)
+				return nil, err
+			}
+			r.Recorder.Event(source, corev1.EventTypeNormal, pingSourceDeploymentCreated, "Cluster-scoped deployment created")
+			return d, nil
+		}
+		return nil, fmt.Errorf("error getting job runner deployment %v", err)
+	}
+	return d, nil
```

Comment:
Could we diff the deployment and update as necessary? For example, the way this is now, it will not pick up any new job images => will not upgrade.